### ISSUE TITLE
Clicking "selected compounds" (RHS) crashes page.

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#766]
https://fragalysis.diamond.ac.uk/viewer/react/projects/1263/923

Initially just diagnose, and work out how much work.

**Comment by ag-m2ms** [comment]
I haven't been able to reproduce the issue. I've tried visiting the page with and without being logged in and both time I was able to open Selected Compunds. Perhaps needs more info (browser, version, etc.)?

**Comment by phraenquex** [comment]
You need to actually select compounds in the RHS.

I see a second bug:  the snapshot did not capture that: 
* the RHS pane was open
* several RHS compounds were selected

![image](https://user-images.githubusercontent.com/33267937/149509243-c62b247a-d65f-4664-8052-9d7307b4a2d3.png)

This is the snapshot of that screenshot - it does not restore for me.  (Firefox.) 
https://fragalysis.diamond.ac.uk/viewer/react/projects/1277/937

**Comment by ag-m2ms** [comment]
The crash should be fixed. The state of open sides (LHS, RHS) is now saved. I wasn't able to reproduce the selection not being saved.

**Comment by NikolaStryckova** [comment]
I tried to reproduce your steps, but during the test, when I decided to display all the side chains, it is not  possible to provide any other action on the RHS (checkboxes are grey). You can see error logs in the console.
![2022-01-24_14h19_51](https://user-images.githubusercontent.com/92084528/150791132-e9792c82-285a-441c-b103-6208bf45f871.png)

**Comment by ag-m2ms** [comment]
Seems like the molecules in RHS miss either molecule_protein or pdf_info field and thus prot_url can't be calculated and used for loading protein. Perhaps an issue with the data? I created a small fix (in the branch #682-fix2) which fixes the crash but I don't think it solves the core issue.

**Comment by phraenquex** [comment]
Yes, this is correct - those data may well be missing - and that is not for this ticket to fix.
The critical thing is that the interface does not crash - handle it gracefully.

(Ideally, it would be possible to report that the data are missing - but this is a different ticket. It's one for Epic-3 - data upload, handling inconsistencies, etc.)

**Comment by phraenquex** [comment]
@NikolaStryckova yes, related to an old problem - fixing it may not be for this different ticket.  (It might even be for a completely different Epic.)

I suggest you create a separate ticket, though, and put it in the Moonshot MVP swimlane - then we can discuss Tuesday whether we fix it or not.

**Comment by NikolaStryckova** [comment]
Thanks for the additional information.  New ticket: #780
